compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -644,6 +646,15 @@ fn print_crate_info(
             TargetSpec => {
                 println!("{}", serde_json::to_string_pretty(&sess.target.to_json()).unwrap());
             }
+            AllTargetSpecs => {
+                let mut targets = BTreeMap::new();
+                for name in rustc_target::spec::TARGETS.to_vec() {
```

Comment:
This `to_vec` seems unnecessary

```suggestion
                for name in rustc_target::spec::TARGETS {
```

Reply:
...right!